blob/parser.go

```diff
@@ -117,19 +117,21 @@ func (p *parser) skipPadding(shares []shares.Share) ([]shares.Share, error) {
 		return nil, errEmptyShares
 	}
 
-	isPadding, err := shares[0].IsPadding()
-	if err != nil {
-		return nil, err
-	}
-
-	if !isPadding {
-		return shares, nil
+	offset := 0
+	for _, sh := range shares {
+		isPadding, err := sh.IsPadding()
+		if err != nil {
+			return nil, err
+		}
+		if !isPadding {
+			break
+		}
+		offset++
 	}
 
-	// update blob index if we are going to skip one share
-	p.index++
-	if len(shares) > 1 {
-		return shares[1:], nil
+	p.index += offset
```

Comment:
is it possible p.index + offset can overflow len shares?

Reply:
no, skipAdding is called when the index is 0 and the offset can't be more than len(shares)